projects/storefrontlib/src/cms-structure/theme/site-theme.service.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  ComponentRef,
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+} from '@angular/core';
+import { SiteThemeConfig } from './site-theme.config';
+
+@Injectable({ providedIn: 'root' })
+export class SiteThemeService {
+  protected rootComponent: ComponentRef<any>;
+  protected renderer: Renderer2;
+
+  constructor(
+    protected config: SiteThemeConfig,
+    protected rendererFactory: RendererFactory2
+  ) {
+    this.renderer = rendererFactory.createRenderer(null, null);
+  }
+
+  init(rootComponent: ComponentRef<any>) {
+    // allow to initialize only once
+    if (!this.rootComponent) {
+      this.rootComponent = rootComponent;
+      this.setTheme(this.config.theme);
+    }
+  }
+
+  setTheme(theme: string) {
+    if (theme) {
+      const element = this.rootComponent.location.nativeElement;
+      // remove the old theme
+      Array.from(element.classList).forEach((attr: string) => {
+        if (attr.startsWith('cx-theme--')) {
```

Comment:
I imagined the CSS rule for the theme is just a global CSS class without any high specificity:
```scss
.alpha {   // note no notion of the root element (it depends on the customer), the class is just global
  --cx-primary-color: blue
}
```

I wanted the theme css class to be unique in the whole storefront page to avoid CSS class name clashes with any other global CSS classes (coming from our Spartacus code, customer's code or any optional UI frameworks).
```scss
.cx-theme--alpha {
  --cx-primary-color: blue
}
```

Reply:
Now I observed we already add such global classes to the storefront component for the page template. Wasn't it a concern before they are global css classes (with low specificity)?

Reply:
We could do cx-storefront.alpha, but we also appreciate customers using a different root component. Customers typically keep the standard app root component and  either import the cx-storefront or introduce their own. 

I'd rather add the prefix in the theming name, and keep the Spartacus logic simple.

Reply:
I'd keep it clean. As said before, we could introduce a prefix, but really i'd either leave it up to the css (`cx-storefront.alpha`) or the naming if the theme.